search-app/src/cmr/search/services/acls/granule_acls.clj

```diff
@@ -152,22 +152,19 @@
         (gc/or-conds [concept-ids-cond entry-titles-cond])
         (or concept-ids-cond entry-titles-cond)))))
 
-
 (defn collection-identifier->query-condition
   "Converts an acl collection identifier to an query condition. Switches implementations based
   on whether the user's query contained collection ids. This implementation assumes the query-coll-ids
   passed in are limited to the provider of the collection identifier."
   [context query-coll-ids provider-id collection-identifier]
   (let [colls-in-prov-cond (provider->collection-condition query-coll-ids provider-id)]
-    (if-let [{:keys [entry-titles access-value temporal]} collection-identifier]
-      (let [entry-titles-cond (provider+entry-titles->collection-condition
-                                context query-coll-ids provider-id entry-titles)
+    (if-let [{:keys [concept-ids entry-titles access-value temporal]} collection-identifier]
+      (let [concept-ids-cond (provider->collection-condition concept-ids provider-id)
             access-value-cond (some-> (access-value->query-condition access-value)
                                       q/->CollectionQueryCondition)
             temporal-cond (some-> temporal temporal->query-condition q/->CollectionQueryCondition)]
 
-        ;; If there's no entry title condition the other conditions must be scoped by provider
-        (gc/and-conds (remove nil? [(or entry-titles-cond colls-in-prov-cond)
+        (gc/and-conds (remove nil? [(or concept-ids-cond colls-in-prov-cond)
```

Comment:
I don't think the logic here is correct. `provider->collection-condition` will just find all collections within the provider when the `concept-ids` is nil. That will always override the `colls-in-prov-cond` and that is not the desired behavior.

Reply:
```(defmulti provider->collection-condition
  "Converts a provider id from an ACL into a collection query condition that will find all collections
  in the provider. query-coll-ids should be a list of the collection ids given by the user in their
  original query (or collection ids resolved from collection query conditions) limited to the
  current provider."
  (fn [query-coll-ids provider-id]
    (if (empty? query-coll-ids)
      :no-query-coll-ids
      :with-query-coll-ids)))

(defmethod provider->collection-condition :no-query-coll-ids
  [query-coll-ids provider-id]
  (q/->CollectionQueryCondition
    (cqm/string-condition :provider-id provider-id true false)))

(defmethod provider->collection-condition :with-query-coll-ids
  [query-coll-ids provider-id]
  (cqm/string-conditions :collection-concept-id query-coll-ids true))
```

Reply:
If concept-ids is nil it uses a provider-id condition.

Reply:
Maybe I should remove the colls-in-prov-cond completely from that line.

Reply:
I think the correct logic is to AND these two collection conditions.